### DISCUSSION TITLE
Method to return whether sqlalchemy will autoincrement a Column or not?

### DISCUSSION THREAD

**Discussion by Dreamsorcerer** [sqlalchemy/sqlalchemy#9261] [Usage Questions]
Is there any way to inspect whether a Column will autoincrement?

I can see `Column.autoincrement`, but it typically returns `"auto"`. I would like to know if SQLAlchemy is going to actually enable autoincrement or not, without having to reproduce all the logic in SQLAlchemy, which appears to require a few checks around `autoincrement`, `primary_key`, `type` and `foreign_keys`.

**Comment by zzzeek** [comment]
We dont have a public method for this but we can make something public, for now what you want to do is look at table._autoincrement_column:

```py
(Pdb) A.__table__._autoincrement_column
Column('id', Integer(), table=<a>, primary_key=True, nullable=False)
(Pdb) A.__table__._autoincrement_column is A.__table__.c.id
True
```


we can add a method like ``table.get_autoincrement_column()`` if there's a use case for this.

**Reply by Dreamsorcerer** [reply]
Sounds good. Use case is aiohttp-admin (which I'm rebuilding). It's a general admin interface, which I want to be able to create from an sqlalchemy model automatically.

To figure out whether an input should be shown in the create form, I want to know if a column is going to be autoincremented or not, and therefore skip the input field for autoincremented columns.

It'd be preferable to have a bool value on the Column itself, but I can certainly make it work with `table.get_autoincrement_column()`.

**Reply by Dreamsorcerer** [reply]
If interested, the relevant code is https://github.com/aio-libs/aiohttp_admin/blob/master/aiohttp_admin/backends/sqlalchemy.py#L43

Have just tested it out with `_autoincrement_column` and that works nicely.
